tests/unit/agent/agent_test.py

```diff
@@ -300,7 +300,7 @@ def test_update_success(mock_model_factory_get):
         # Capture warnings
         with pytest.warns(
             DeprecationWarning,
-            match="update\(\) is deprecated and will be removed in a future version. Please use save\(\) instead.",
+            match="update\(\) is deprecated and will be removed in a future version. Please use save\(\) instead.",  # noqa: W605
```

Comment:
Remove the qa comment.

Instead of match, is there a way to just check for a substring? Then we can remove the part that flake8 complains.